set.mm

```diff
@@ -465114,6 +465132,141 @@ its graph has a given second element (that is, function value).
 
 $( End $[ set-mbox-ta-xdiv.mm $] $)
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Words over a set - misc additions
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  ${
+    wrdfd.n $e |- ( ph -> N = ( # ` W ) ) $.
+    wrdfd.w $e |- ( ph -> W e. Word S ) $.
+    $( A word is a zero-based sequence with a recoverable upper limit,
+       deduction version.  (Contributed by Thierry Arnoux, 22-Dec-2021.) $)
+    wrdfd $p |- ( ph -> W : ( 0 ..^ N ) --> S ) $=
+      ( cc0 cfzo co wf chash cfv cword wcel wrdf syl oveq2d feq2d mpbird ) AGCH
+      IZBDJGDKLZHIZBDJZADBMNUCFBDOPATUBBDACUAGHEQRS $.
+  $}
+
+  $( Condition for the restriction of a word to be a word itself.  (Contributed
+     by Thierry Arnoux, 5-Oct-2018.) $)
+  wrdres $p |- ( ( W e. Word S /\ N e. ( 0 ... ( # ` W ) ) )
+    -> ( W |` ( 0 ..^ N ) ) e. Word S ) $=
+    ( cword wcel cc0 chash cfv cfz co cfzo cres wss wrdf cuz elfzuz3 fzoss2 syl
+    wa wf fssres syl2an iswrdi ) CADZEZBFCGHZIJEZSFBKJZACUHLZTZUIUDEUEFUFKJZACT
+    UHUKMZUJUGACNUGUFBOHEULBFUFPBFUFQRUKAUHCUAUBABUIUCR $.
+
+  ${
+    $d v N $.  $d v S $.  $d v W $.
+    $( Existence of a split of a word at a given index.  (Contributed by
+       Thierry Arnoux, 11-Oct-2018.)  (Proof shortened by AV, 3-Nov-2022.) $)
+    wrdsplex $p |- ( ( W e. Word S /\ N e. ( 0 ... ( # ` W ) ) )
+      -> E. v e. Word S W = ( ( W |` ( 0 ..^ N ) ) ++ v ) ) $=
+      ( cword wcel chash cfv cop csubstr co cc0 cfz cfzo cres cconcat wceq wrex
+      cv cpfx swrdcl wa cuz simpr elfzuz2 eluzfz2 ccatpfx mpd3an3 pfxres oveq1d
+      3syl pfxid adantr 3eqtr3rd oveq2 rspceeqv syl2an2r ) DBEZFZDCDGHZIJKZURFC
+      LUTMKZFZDDLCNKOZVAPKZQDVDASZPKZQAURRBDCUTUAUSVCUBZDCTKZVAPKZDUTTKZVEDUSVC
+      UTVBFZVJVKQVHVCUTLUCHFVLUSVCUDCLUTUELUTUFUKBDCUTUGUHVHVIVDVAPBDCUIUJUSVKD
+      QVCBDULUMUNAVAURVGVEDVFVAVDPUOUPUQ $.
+
+    $( Obsolete proof of ~ wrdsplex as of 12-Oct-2022.  (Contributed by Thierry
```

Comment:
I think this OLD theorem can be deleted now.

Reply:
Yes!